app/src/main/java/org/feature/fox/coffee_counter/ui/CoreActivity.kt

```diff
@@ -41,5 +52,78 @@ class CoreActivity : ComponentActivity() {
                 }
             }
         }
+        mNfcAdapter = NfcAdapter.getDefaultAdapter(this)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        if (mNfcAdapter != null) {
+            val options = Bundle()
+            // Work around for some broken Nfc firmware implementations that poll the card too fast
+            options.putInt(NfcAdapter.EXTRA_READER_PRESENCE_CHECK_DELAY, 250)
+
+            // Enable ReaderMode for all types of card and disable platform sounds
+            mNfcAdapter!!.enableReaderMode(
```

Comment:
Lars pointed out that we shouldn't work with the `!!`-operator.
The reason for that is that after the null-check it could still turn into null if i understood that right.